### ISSUE TITLE
updateqQueries on query

### ISSUE THREAD

**Issue by tomitrescak** [apollostack/apollo-client#606]
Hi

I am trying to create a nice polling mechanism on large amount of data. 
Currently I have following:

1. Query fetches a large amount of data
2. Mutation checks which fields have been updated and retrieves them. Using `updateQueries` I can modify my store

The thing here is, why do I have to use mutation to do this? Would be great if I could control the result of the query and decide how it will be merged in the store.

Long story short, would be great to have `updateQueries` also on a query.

**Comment by helfer** [comment]
Have you tried using `fetchMore`?

**Comment by tomitrescak** [comment]
Now that's the answer I love ... exactly what I need ;) Sorry for not reading properly the documentation.

**Comment by tomitrescak** [comment]
One detail though ... how would I use this with apollo-react? Do I have access to the observer ?

**Comment by stubailo** [comment]
You get `fetchMore` in the same places you get `refetch`.

**Comment by tomitrescak** [comment]
Do you reckon in "props" method of the Apollo-react query is a good place?

**Comment by stubailo** [comment]
Yes, that's probably the right place to define a function to load another bit of data!

**Comment by tomitrescak** [comment]
Thanks sashko.

**Comment by tomitrescak** [comment]
There is something WRONG with trying to do it in "props". What is happening is that after every refresh, my component will re-render exponentially amount of times. First, 2x, second 4, third 8x ... you get the idea. In the end I end up with fetchmore called n^2 times. Not sure why. Do you see anything terribly wrong with the following?

```js
const withSolutions = graphql(gql`
  query markingSolutions($practicalId: String, $semesterId: String, $lastModification: Date, $userId: String) {
    markingSolutions(semesterId: $semesterId, practicalId: $practicalId, lastModification: $lastModification, userId: $userId) {
      _id
      user
      userId
      questionId
      practicalId
      semesterId
      exerciseId
      userQuestion
      userAnswer
      tutorComment
      finished
      modified
      mark
    }
  }
`,
  {
    name: 'solutionData',
    options: (ownProps: IComponent) => ({
      variables: {
        practicalId: ownProps.params.practicalId,
        semesterId: ownProps.params.semesterId,
        userId: ownProps.userId,
        lastModification: ownProps.selectedDate
      }
    }),
    props: ({ solutionData, ownProps }: any) => {
      // in 2 seconds fetch more data
      if (solutionData.fetchMore) {
        setTimeout(() => {
          console.log("Fetching data ...");
          solutionData.fetchMore({
            // fetch 10 more items starting from new offset
            // limit and id variables are the same
            variables: {
              practicalId: ownProps.params.practicalId,
              semesterId: ownProps.params.semesterId,
              userId: ownProps.userId,
              lastModification: getLastModification()
            },
            // tell Apollo Client how to merge the new results of the query
            updateQuery: (previousResult: any, { fetchMoreResult, queryVariables }: any) => {
              const prevSolutions = previousResult;
              const newSolutions = fetchMoreResult.markingSolutions; // I'm not merging yet
              const newResult = Object.assign({}, prevSolutions);

              // now browse 

              return newResult;
            }
          });
        }, 5000);
      }

      return {
        solutions: solutionData.markingSolutions
      }
    }
  });
```

**Comment by tomitrescak** [comment]
And of course immediatelly after posting I realised my error. I was not clearing the timeout in case multiple updates arrived.
